keras/utils/metrics_utils.py

```diff
@@ -839,3 +839,123 @@ def _assert_splits_match(nested_splits_lists):
       for splits_list in nested_splits_lists[1:]
       for (s1, s2) in zip(nested_splits_lists[0], splits_list)
   ]
+
+
+def binary_matches(y_true, y_pred, threshold=0.5):
+  """Creates int Tensor, 1 for label-prediction match, 0 for mismatch.
+
+  Args:
+    y_true: Ground truth values. shape = `[batch_size, d0, .. dN]`.
+    y_pred: The predicted values. shape = `[batch_size, d0, .. dN]`.
+    threshold: (Optional) Float representing the threshold for deciding whether
+      prediction values are 1 or 0.
+
+  Returns:
+    Binary matches. shape = `[batch_size, d0, .. dN]`
+  """
+  y_pred = tf.convert_to_tensor(y_pred)
+  threshold = tf.cast(threshold, y_pred.dtype)
+  y_pred = tf.cast(y_pred > threshold, y_pred.dtype)
+  return tf.cast(tf.equal(y_true, y_pred), backend.floatx())
+
+
+def sparse_categorical_matches(y_true, y_pred):
+  """Creates float Tensor, 1.0 for label-prediction match, 0.0 for mismatch.
+
+  You can provide logits of classes as `y_pred`, since argmax of
+  logits and probabilities are same.
+
+  Args:
+    y_true: Integer ground truth values.
+    y_pred: The prediction values.
+
+  Returns:
+    Match tensor: 1.0 for label-prediction match, 0.0 for mismatch.
+  """
+  y_pred = tf.convert_to_tensor(y_pred)
+  y_true = tf.convert_to_tensor(y_true)
+  y_pred_rank = y_pred.shape.ndims
+  y_true_rank = y_true.shape.ndims
+  # If the shape of y_true is (num_samples, 1), squeeze to (num_samples,)
+  if (y_true_rank is not None) and (y_pred_rank is not None) and (len(
+      backend.int_shape(y_true)) == len(backend.int_shape(y_pred))):
+    y_true = tf.squeeze(y_true, [-1])
+  y_pred = tf.math.argmax(y_pred, axis=-1)
+
+  # If the predicted output and actual output types don't match, force cast them
+  # to match.
+  if backend.dtype(y_pred) != backend.dtype(y_true):
+    y_pred = tf.cast(y_pred, backend.dtype(y_true))
+  return tf.cast(tf.equal(y_true, y_pred), backend.floatx())
+
+
+def categorical_matches(y_true, y_pred):
```

Comment:
I am not sure this utility is needed, you could just inline its contents where it is used.

Reply:
Just checking to make sure I understand.
Do you mean to delete this utility method and where categorical_matches is called just call sparse_categorical_matches but argmax the y_true when passing it into the sparse method? Sorry if I'm misinterpreting.

Reply:
Yes, exactly.